### ISSUE TITLE
One problem remains to be solved

### ISSUE THREAD

**Issue by karesake** [samemory/homebridge-eufy-security#367]
### One problem remains to be solved

Now that the video live stream is working on Apple devices, there is still a problem on Apple TV and Apple Watch that needs to be solved.

The stream always starts with cameras - except for the doorbell - only on the second attempt.

The first time the stream loads and the camera shows "No Response". Directly after that, if you start the stream again, it loads immediately. _And it doesn't matter whether you start the stream on your iPhone, iPad, Apple Watch, AppleTV or Mac. On the first try: no answer and immediately after that it works..._

This is the only problem regarding the live stream that still needs to be solved. Because especially on the Apple TV it's not fun otherwise.

**@lenoxys do you have an idea?**

I don't think it has anything to do with the resolution or the stream in general anymore. I think the fault must be somewhere else.

I also don't think it has anything to do with waking up the camera. If I keep the stream running permanently on the mac, for example, the stream on the iPhone only starts on the second attempt...

**What we should do on the way to a perfect plugin**

**✔  More than 300 seconds streaming time** 
_What we should definitely take out is the 300 seconds maximum streaming time. Either 30 minutes or Unlimited. Otherwise the stream will stop after 3 minutes on the ATV._

**✔  See that the stream loads on the first try.** 

_______________________________________________________________

**EDIT 22/03/21**

OK. I think (and hope) it's solved.

I will make a pull request **if everything runs smoothly for a few days.**

**Comment by r33ce15** [comment]
Interested to try this I so far have had eufy settings to low compression and smart display mode and then using copy function in ffmpeg and it's been working smoothly get snapshots and live stream loading fast but every now and then stops working in both eufy app and home but this could be network related have to see how goes but this interesting find!

**Comment by spextec** [comment]
@karesake  

Thank you very much for being so committed to ensuring that streaming works well. I installed your fork and the cameras work perfectly. I also hope that it stays that way. The stream starts in less than 4 seconds and always on the first attempt. Thank you!

**Comment by schliemann** [comment]
@karesake Great work. When would it make sense to make a PR. I can see you are making a lot of commits.

**Comment by karesake** [comment]
@schliemann I don't know how i can create a pull request. I have to deal with that a little more. And yes, since I'm actually the only one using my fork, there are many changes.


But at the current stage, I can say that the stream works without any problems - on all devices.

(1) For example, I have since removed the Local Livestream. With the latest Eufy client, the camera stream builds up just as easily (and faster). The doorbell stream now also works on the AppleTV.

(2) Even if the bell rings, the live image will be displayed within 3 seconds.

(3) Furthermore, the stream is renewed automatically. For example, if HomeKit shows "No Response" on the first try ... especially on the Apple TV ... the stream automatically reconnects, almost without interruption.

**But I'll make the pull request once I've tested the whole thing for a few days and it's running smoothly.**

**Comment by lenoxys** [comment]
Thanks men. Let me know if you are not able to push a PR. I will do it on my side and credit your awesome work.

**Comment by karesake** [comment]
Let's have a look. If I can't do it that way, feel free to use everything and create it yourself.

BTW: Apple Watch Stream is working 😁

<a href="https://ibb.co/ZN4tV1K"><img src="https://i.ibb.co/ZN4tV1K/A9-DF1-BC8-D2-EA-47-C7-9151-A77306-AB52-A8.jpg" alt="A9-DF1-BC8-D2-EA-47-C7-9151-A77306-AB52-A8" border="0"></a> <a href="https://ibb.co/Cz2ZcZp"><img src="https://i.ibb.co/Cz2ZcZp/4716588-D-60-B9-4-A84-9-A55-3-F5-B3-F8-EE28-C.jpg" alt="4716588-D-60-B9-4-A84-9-A55-3-F5-B3-F8-EE28-C" border="0"></a>

**Comment by karesake** [comment]
I think I am currently at a point to offer my version of the measurements. The version is primarily aimed at a fast and reliable stream on all Apple devices.

@lenoxys @schliemann 

I don't know what to do with a pull request. You can simply take the data on my fork and create one yourself and of course test it beforehand.

• Snapshots are created quickly and automatically renewed after about a minute.
• The stream works on all Apple devices.
• The stream will be built up for me within 3 seconds

**Comment by schliemann** [comment]
@karesake @lenoxys We should maybe create a new feature branch and PR it in there first so we can do a quick smoke test.

**Comment by fullphat** [comment]
Does this only apply to the doorbell cameras or to all?

**Comment by fullphat** [comment]
> I think I am currently at a point to offer my version of the measurements. The version is primarily aimed at a fast and reliable stream on all Apple devices.
> 
> @lenoxys @schliemann
> 
> I don't know what to do with a pull request. You can simply take the data on my fork and create one yourself. Please test it yourself beforehand. My doorbells and non-HomeKit cameras all work perfectly.
> 
> • Snapshots are created quickly and automatically renewed after about a minute. • The stream works on all Apple devices. • The stream will be built up for me within 3 seconds • The Vcodes does not have to be set to Copy. • FFMPEG Debug does not need to be activated.

Does the above apply to all cameras, or just to the doorbell cam range?

**Comment by lenoxys** [comment]
Before going master I'd like to allow you to custom per camera. But we'll use @karesake default value for those who don't know what to put on it.

**Comment by karesake** [comment]
However, making the whole thing adaptable to individual cameras is of no use, since HomeKit always uses the same thing.

iPhone, iPad: 1280x720, 30FPS, 299KB/S
Apple TV, iPad Pro Landscape: 1920x1080, 30FPS, 802KB/S

If you want to make it customizable, it would be interesting if the RTSP can be used as the source.

**Comment by lenoxys** [comment]
ok just implemented the code. needs to be documented

**Comment by karesake** [comment]
Can we put in the RTSP Stream now ? 

Actually don't Access to Mac HomeBridge :(

**Comment by lenoxys** [comment]
this new beta broke some settings which was global and now per camera such as `enableButton`, `motionButton`, `codec` and `ffmpegdebug`.

No I have not worked on RTSP this is a tricky one since some camera doesn't support it.

**Comment by lenoxys** [comment]
need to add yours change as well. I like the approach but I'm not sure is a good idea to provide conf per platform I think per camera is enough and if you change some it will apply whatever the platform is. (mainly use I think for debug)

**Comment by jjr3211** [comment]
Updated to the new beta and it looks like it broke my doorbell camera?

**Comment by karesake** [comment]
> Updated to the new beta and it looks like it broke my doorbell camera?

Confirm that. Doorbell and author Cameras not not working anymore. And I don't see any setting options for this either. Went back to my Version and everything is fine.

**Comment by lenoxys** [comment]
Manage to add RTSP look at the latest beta (need to enable RTSP stream first on Eufy App)

**Comment by lenoxys** [comment]
![image](https://user-images.githubusercontent.com/3996456/160896665-64bea85c-a466-4771-9235-6b3ee2b754f0.png)
Working on the config...

**Comment by karesake** [comment]
That looks great.


**I already have the following in another post** 

_"Unfortunately, RTSP doesn't really work. As with Camera UI, the stream is no longer accessible after a few minutes"_


**I also get these messages:**


_[homebridge-eufy-security] This plugin slows down Homebridge. The write handler for the characteristic 'Selected RTP Stream Configuration' on the accessory 'Überwachung Garage 34BD' didn't respond at all!. Please check that you properly call the callback! See https://git.io/JtMGR for more info._

**and**

_Überwachung Garage (FloodLight Cam 2 pro) handleMotionDetectedGet Wrong return value_

**Comment by lenoxys** [comment]
> That looks great.
> 
> **I already have the following in another post**
> 
> _"Unfortunately, RTSP doesn't really work. As with Camera UI, the stream is no longer accessible after a few minutes"_
> 

For that point, we won't make much than a plugin which have the best dev  `¯\_(ツ)_/¯`

> **Without RTSP**
> 
> As before, the camera needs two attempts to establish the stream. He breaks the first time
> 
> The funny thing is, if I turn off motion detection in the Eufy app, it works. I think it has something to do with that.

Good catch !

> 
> **I also get these messages:**
> 
> _[homebridge-eufy-security] This plugin slows down Homebridge. The write handler for the characteristic 'Selected RTP Stream Configuration' on the accessory 'Überwachung Garage 34BD' didn't respond at all!. Please check that you properly call the callback! See https://git.io/JtMGR for more info._
> 
> **and**
> 
> _Überwachung Garage (FloodLight Cam 2 pro) handleMotionDetectedGet Wrong return value_

let's create another thread to keep it clear for each issue your facing. Need to organise because it's really hard to follow with generic title
